lib/vehicle.rb

```diff
@@ -22,4 +26,19 @@ def antique?
   def electric_vehicle?
     @engine == :ev
   end
+
+  def set_date
+    @registration_date = Date.today
+  end
+
+  def give_plate
+    if engine == :ev
+      @plate_type = :ev
+    elsif @engine == :ice && Date.today.year - @year > 25
+      @plate_type = :antique
+    else
+      @plate_type = :regular
+    end
+  end
```

Comment:
Remember to look at the methods you already have to avoid repeating code. We had the methods `.electric_vehicle?` and `.antique?` that we could have used for the conditions of the flow control.

Reply:
This was another element that I saw in other people's projects that looks cleaner and more readable. I will keep my eye out for that in the future.